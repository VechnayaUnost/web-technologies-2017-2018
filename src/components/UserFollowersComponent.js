import React from 'react';
import { connect } from "react-redux";
import { bindActionCreators } from 'redux';
import { Link } from 'react-router-dom';
import { fetchUserFollowersRequest } from '../actions'

class UserFollowersComponent extends React.Component {
    componentDidMount() {
        this.props.fetchUserFollowersRequest(this.props.userLogin);
    }

    render() {
        const followers = [];
        for (const follower in this.props.followers) {
            if (this.props.followers.hasOwnProperty(follower)) {
                const element = this.props.followers[follower];
                followers.push(element);
            }
        }
        return (
            this.props.userLogin === '' || this.props.userLogin === undefined
            ?
            <div>
                <ul>
                    <li>
                        <Link to="/user/followers">User followers</Link>
                    </li>
                    <li>
                        <Link to="/user/repos">User repositories</Link>
                    </li>
                </ul>

                <h2>Followers</h2>
                <h3>Empty</h3>
            </div>
            :
            <div>
                <ul>
                    <li>
                        <Link to="/user/followers">User followers</Link>
                    </li>
                    <li>
                        <Link to="/user/repos">User repositories</Link>
                    </li>
                </ul>
                <h2>Followers</h2>
                {followers.map(follower =>
                    <div>
                        <h3>{follower.login}</h3>
                        <img src={follower.avatar_url} alt={'user avatar'}/>
                    </div>
                )}
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        userLogin: state.userInfo.login,
        followers: state.userFollowers
    };
};

const mapDispatchToProps = dispatch => bindActionCreators({
    fetchUserFollowersRequest,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(UserFollowersComponent);