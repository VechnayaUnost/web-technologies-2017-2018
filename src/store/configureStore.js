import { createStore, applyMiddleware } from 'redux';
import createSagaMiddleware, { END } from 'redux-saga';

import rootReducer from '../reducers';
import rootSaga from '../sagas';

export default (initialState) => {
    const sagaMiddleware = createSagaMiddleware();
    const store = createStore(rootReducer, initialState, applyMiddleware(sagaMiddleware));

    sagaMiddleware.run(rootSaga);
    store.runSaga = () => sagaMiddleware.run(rootSaga);
    store.close = () => store.dispatch(END);

    return store;
};