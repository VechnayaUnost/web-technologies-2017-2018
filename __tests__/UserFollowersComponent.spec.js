import React from 'react';
import UserFollowersComponent from '../src/components/UserFollowersComponent';
import { shallow, mount, configure } from 'enzyme';
import { shallowToJson, mountToJson } from 'enzyme-to-json';
import { BrowserRouter } from 'react-router-dom';
import Adapter from 'enzyme-adapter-react-16';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });

describe('UserFollowers component test', () => {
    const mockStore = configureStore([ thunk ]);
    const initialState = {
        userInfo: {
            login: 'GoGoPandora'
        },
        userFollowers: [
            {
                login: 'FakeAvatar1337',
                avatar_url: 'https://vokrug.tv/pic/person/2/3/8/8/23887ed0f336731e4c62eedcb25bf4d0.jpeg'
            }
        ]
    }

    let store;

    beforeEach(() => {
        store = mockStore(initialState);
    })

    it('UserFollowers component rendered succesfully', () => {
        const userFollowers = mount(
            <BrowserRouter>
                <UserFollowersComponent store={store} />
            </BrowserRouter>
        );
        expect(mountToJson(userFollowers)).toMatchSnapshot();
    });

    it('UserFollowers rendered without user login', () => {
        const fakeInitialState = {
            userInfo: {
                login: ''
            },
            userFollowers: [
                {
                    login: 'FakeAvatar1337',
                    avatar_url: 'https://vokrug.tv/pic/person/2/3/8/8/23887ed0f336731e4c62eedcb25bf4d0.jpeg'
                }
            ]
        }
        const fakeStore = mockStore(fakeInitialState);
        const userFollowers = mount(
            <BrowserRouter>
                <UserFollowersComponent store={fakeStore} />
            </BrowserRouter>
        );
        expect(userFollowers.prop('userLogin')).toBe(undefined);
        expect(mountToJson(userFollowers)).toMatchSnapshot();
    })
});