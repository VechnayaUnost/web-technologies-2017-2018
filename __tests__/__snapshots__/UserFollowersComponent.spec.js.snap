// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserFollowers component test UserFollowers component rendered succesfully 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(UserFollowersComponent)
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <UserFollowersComponent
        fetchUserFollowersRequest={[Function]}
        followers={
          Array [
            Object {
              "avatar_url": "https://vokrug.tv/pic/person/2/3/8/8/23887ed0f336731e4c62eedcb25bf4d0.jpeg",
              "login": "FakeAvatar1337",
            },
          ]
        }
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        storeSubscription={
          Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
        }
        userLogin="GoGoPandora"
      >
        <div>
          <ul>
            <li>
              <Link
                replace={false}
                to="/user/followers"
              >
                <a
                  href="/user/followers"
                  onClick={[Function]}
                >
                  User followers
                </a>
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/user/repos"
              >
                <a
                  href="/user/repos"
                  onClick={[Function]}
                >
                  User repositories
                </a>
              </Link>
            </li>
          </ul>
          <h2>
            Followers
          </h2>
          <div>
            <h3>
              FakeAvatar1337
            </h3>
            <img
              alt="user avatar"
              src="https://vokrug.tv/pic/person/2/3/8/8/23887ed0f336731e4c62eedcb25bf4d0.jpeg"
            />
          </div>
        </div>
      </UserFollowersComponent>
    </Connect(UserFollowersComponent)>
  </Router>
</BrowserRouter>
`;

exports[`UserFollowers component test UserFollowers rendered without user login 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Connect(UserFollowersComponent)
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <UserFollowersComponent
        fetchUserFollowersRequest={[Function]}
        followers={
          Array [
            Object {
              "avatar_url": "https://vokrug.tv/pic/person/2/3/8/8/23887ed0f336731e4c62eedcb25bf4d0.jpeg",
              "login": "FakeAvatar1337",
            },
          ]
        }
        store={
          Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
        }
        storeSubscription={
          Subscription {
            "listeners": Object {
              "clear": [Function],
              "get": [Function],
              "notify": [Function],
              "subscribe": [Function],
            },
            "onStateChange": [Function],
            "parentSub": undefined,
            "store": Object {
              "clearActions": [Function],
              "dispatch": [Function],
              "getActions": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
            },
            "unsubscribe": [Function],
          }
        }
        userLogin=""
      >
        <div>
          <ul>
            <li>
              <Link
                replace={false}
                to="/user/followers"
              >
                <a
                  href="/user/followers"
                  onClick={[Function]}
                >
                  User followers
                </a>
              </Link>
            </li>
            <li>
              <Link
                replace={false}
                to="/user/repos"
              >
                <a
                  href="/user/repos"
                  onClick={[Function]}
                >
                  User repositories
                </a>
              </Link>
            </li>
          </ul>
          <h2>
            Followers
          </h2>
          <h3>
            Empty
          </h3>
        </div>
      </UserFollowersComponent>
    </Connect(UserFollowersComponent)>
  </Router>
</BrowserRouter>
`;
